data/moves.js

```diff
@@ -16438,19 +16438,26 @@ let BattleMovedex = {
 			noCopy: true,
 			onStart(target) {
 				this.effectData.layers = 1;
+				this.effectData.def = 0;
+				this.effectData.spd = 0;
 				this.add('-start', target, 'stockpile' + this.effectData.layers);
+				let [curDef, curSpD] = [target.boosts.def, target.boosts.spd];
 				this.boost({def: 1, spd: 1}, target, target);
+				if (curDef !== target.boosts.def) this.effectData.def--;
+				if (curSpD !== target.boosts.spd) this.effectData.spd--;
 			},
 			onRestart(target) {
 				if (this.effectData.layers >= 3) return false;
 				this.effectData.layers++;
 				this.add('-start', target, 'stockpile' + this.effectData.layers);
+				let [curDef, curSpD] = [target.boosts.def, target.boosts.spd];
```

Comment:
...there is no need for destructuring here; these should be two separate variable declarations.

Allocating the array is unnecessary.

```suggestion
				const curDef = target.boosts.def;
				const curSpD = target.boosts.spd;
```